napari/layers/utils/array_interface.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from typing import Tuple, Union
+
+import numpy as np
+
+if sys.version_info.minor < 8:
+    from typing_extensions import Protocol
+else:
+    from typing import Protocol
+
+
+class ArrayInterface(Protocol):
+    def __getitem__(self, item) -> Union['ArrayInterface', int, float, bool]:
+        ...
+
+    def __len__(self) -> int:
+        ...
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return (0,)
+
+    @property
+    def dtype(self) -> type:
+        return np.uint8
```

Comment:
interestingly enough, these two actually _shouldn't_ be `@property` ... since they _are_ editable in a numpy array.  try it out:
```python
In [42]: x = np.ones((10,10))                                                                                                                                                                  

In [43]: x.shape                                                                                                                                                                               
Out[43]: (10, 10)

In [44]: x.shape = (25, 4)                                                                                                                                                                     

In [45]: x.shape                                                                                                                                                                               
Out[45]: (25, 4)

In [46]: x                                                                                                                                                                                     
Out[46]: 
array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
...
```

Reply:
Now I remember.. Only assumption is that new shape need to multiply to same size. And type overwrite change size.